invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js

```diff
@@ -5,32 +5,136 @@
 // under the terms of the MIT License; see LICENSE file for more details.
 
 import PropTypes from "prop-types";
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { i18next } from "@translations/invenio_administration/i18next";
 import CopyButton from "./CopyButton";
-import { Table, Header } from "semantic-ui-react";
+import { Table, Header, Dropdown, Grid } from "semantic-ui-react";
+import { http } from "@js/invenio_administration/src/api/config.js";
+import { capitalize } from "lodash";
+
+/** Map of known formats and their name. */
+const knownFormats = {
+  "oai_dc": "OAI Dublin Core",
+  "datacite": "DataCite",
+  "oai_datacite": "OAI DataCite",
+}
 
 function LinksTable({ data }) {
+  const [links, setLinks] = useState(data.links);
+  const [formats, setFormats] = useState([]);
+
+  useEffect(async () => {
+    try {
+      const response = await http.get(
+        '/api/oaipmh/formats'
+      );
+      const formats = response.data?.hits?.hits;
+      if (Array.isArray(formats) && formats.length > 0) {
+        const serialized = formats.map((formt) => {
+          return {
+            key: formt.id,
+            value: formt.id,
+            text: knownFormats[formt.id] ?? formatKeyToName(formt.id)
+          }
+        });
+        setFormats(serialized);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }, []);
+
+  /**
+   * Replaces the metadata prefix in the link.
+   * Used to update links when the metadata prefix is changed.
+   */
+  const replaceLinkPrefix = (link, newPrefix) => {
+    const oldPrefix = getPrefixFromLink(link);
+    if (oldPrefix) {
+      return link.replace(oldPrefix, newPrefix)
+    }
+    return link;
+  };
+
   /**
-   * WIP: this method should remove metadataPrefix from the url once the attribute is listed
-   * as a dropdown.
+   * Retrieves metadata prefix from a link.
+   * Uses a regex to retrieve the prefix, matching "&metadataPrefix=...&".
    *
-   * @param {string} url
-   * @returns {string} stripped url
+   * Returns null if pattern not found.
    */
-  const sanitizeLink = (url) => {
-    return url;
+  const getPrefixFromLink = (link) => {
+    const prefixRegex = /\&metadataPrefix\=(.*)\&/;
```

Comment:
Decided to use a regex so the component only has one state that triggers re-renders (`links`).

Initially was using a state `currentPrefix` to keep track of the prefix currently being used but that required two states that, after being updated, would trigger a re-render each.

The downside of this approach is that it's tied to the backend's `&metadataPrefix&` parameter. If that changes in the future, this will not work properly.

If I am mistaken and having two states would be better / cleaner, please let me know and this can be modified.

Reply:
Thanks for the clarification! I personally tend to choose the option that is more readable, so I would probably go for the `currentPrefix` option, but this is just personal choice, this looks good to me. :+1:

Reply:
Future change:
agree, please go with the state to make this component easier to understand and maintain. We are not very worried here about re-rendering, given that this will happen after a user action.
Let's parse the URL with [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) and manipulate query args with [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams), avoiding `RegEx`.